dbms/src/Common/UTF8Helpers.h

```diff
@@ -49,9 +56,46 @@ inline size_t seqLength(const UInt8 first_octet)
 inline size_t countCodePoints(const UInt8 * data, size_t size)
 {
     size_t res = 0;
+    const auto end = data + size;
+
+#if __SSE2__ && __SSE4_1__
+    const auto bytes_sse = sizeof(__m128i);
+    const auto src_end_sse = (data + size) - (size % bytes_sse);
+
+    const auto one_sse = _mm_set1_epi8(1);
+    const auto zero_sse = _mm_set1_epi8(0);
+
+    const auto align_sse = _mm_set1_epi8(0xFF - 0xC0);
+    const auto upper_bound = _mm_set1_epi8(0x7F + (0xFF - 0xC0) + 1);
+
+    for (; data < src_end_sse;)
+    {
+        auto sse_res = _mm_set1_epi8(0);
+
+        for (int i = 0, limit = 0xFF; i < limit && data < src_end_sse; ++i, data += bytes_sse)
+        {
+            const auto chars = _mm_loadu_si128(reinterpret_cast<const __m128i *>(data));
+
+            ///Align to zero for the solve two case
+            const auto align_res = _mm_add_epi8(chars, align_sse);
+            ///Because _mm_cmmpgt_epid8 return 0xFF if true
+            const auto choose_res = _mm_blendv_epi8(zero_sse, one_sse, _mm_cmpgt_epi8(align_res, upper_bound));
+
+            sse_res = _mm_add_epi8(sse_res, choose_res);
+        }
+
+        UInt16 mem_res[8] = {0};
```

Comment:
@alexey-milovidov  Sorry, I flushed your comments.
Yes, I'm not sure it's better.Can you give me some advice? Thanks